assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -204,42 +210,97 @@ const baseActions = {
 
 		// Sync available custom dimensions.
 		if ( missingCustomDimensions.length > 0 ) {
-			yield baseActions.syncAvailableCustomDimensions( propertyID );
+			yield baseActions.syncAvailableCustomDimensions();
 		}
+
+		yield {
+			type: SET_CUSTOM_DIMENSIONS_BEING_CREATED,
+			payload: { customDimensions: [] },
+		};
 	},
 
 	/**
 	 * Syncs available custom dimensions in the settings.
 	 *
 	 * @since n.e.x.t
 	 *
-	 * @param {string} propertyID GA4 property ID.
 	 * @return {Array<string>} Available custom dimensions.
 	 */
-	syncAvailableCustomDimensions: createValidatedAction(
-		( propertyID ) => {
-			invariant(
-				isValidPropertyID( propertyID ),
-				'A valid GA4 propertyID is required.'
+	*syncAvailableCustomDimensions() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const propertyID = registry
+			.select( MODULES_ANALYTICS_4 )
+			.getPropertyID();
+
+		yield {
+			type: SET_AVAILABLE_CUSTOM_DIMENSIONS_SYNCING_ACTION,
+			payload: { isSyncing: true },
+		};
+
+		const { response, error } =
+			yield fetchSyncAvailableCustomDimensionsStore.actions.fetchSyncAvailableCustomDimensions(
+				propertyID
```

Comment:
No need to pass the propertyID parameter anymore.

Reply:
Updated, thanks!